Realm/Realm.Weaver/RealmWeaver.cs

```diff
@@ -390,10 +402,18 @@ private WeavePropertyResult WeaveProperty(PropertyDefinition prop, TypeDefinitio
             {
                 var genericArguments = ((GenericInstanceType)prop.PropertyType).GenericArguments;
                 var elementType = genericArguments.Last();
-                if (!elementType.Resolve().IsValidRealmObjectBaseInheritor(_references) &&
-                    !_realmValueTypes.Contains(elementType.FullName))
+
+                if (!elementType.Resolve().IsValidRealmObjectBaseInheritor(_references))
                 {
-                    return WeavePropertyResult.Error($"{type.Name}.{prop.Name} is an {collectionType} but its generic type is {elementType.Name} which is not supported by Realm.");
+                    if (_counterTypes.Contains(elementType.FullName))
```

Comment:
Rather than having a new list of counters, you can check if the type is realm integer by calling `elementType.Resolve().IsRealmInteger(out _, out _)`.